### ISSUE TITLE
Great work, One Suggestion!

### ISSUE THREAD

**Issue by NJseo** [morassman/atom-commander#27]
Thank you for committing your time to this project, its great. I was wondering or hoping that you would add Chmod permissions to the options that are available.

Also, what would be cool is the palette was dockable to other windows.

**Comment by morassman** [comment]
I do want to add chmod permissions. The other day I actually needed it myself.

I'm not 100% sure what you mean with "dockable to other windows" though. Do you mean other parts of the window, as in, top, left and right?

**Comment by NJseo** [comment]
I run two monitors, sometimes three. Being able to take the commander pallette and move it to another monitor would be great.

**Comment by morassman** [comment]
Ok I understand now. Dragging panels from one Atom window to another would be quite an undertaking. Atom supports dragging tabs, but not panels. Also, each Atom window already has its own instance of the Atom Commander panel, so moving the panel between windows doesn't really make sense.

I guess what it comes down to is that you'd like the AC panel in the one window to display what the AC panel in another window is showing. In that case, dragging the panel from one window to another can then simply cause the target panel to navigate to the folders being shown by the source panel. You wouldn't actually be dragging the panel itself, but just it's state.

Dragging the state in this way would be possible. I must admit though, I'm not _super_ keen to add this, but I'll take a look at how it can be done.

**Comment by NJseo** [comment]
Geez, I don't know. I use multiple screens to do development work and being able to organize these blocks or palettes across my various screens would be helpful. I am still struggling to get accustomed to Atom. I tell it is a feature that I miss from my Dreamweaver days, that and simplified menus, including context menus with options for commenting selected code.

**Comment by morassman** [comment]
Yeah. I think I can relate a bit. I'm used to developing in Eclipse and organising my workspace any way I like. It would be nice if Atom was more flexible in that regard.

Ok. So dragging the panel between windows isn't something I'm going to do now, but I'll see what I can do about syncing the state of an AC panel with one in another window.

**Comment by NJseo** [comment]
Far enough, do you have any suggestions for a context menu in the code window with useful tools in it?

**Comment by morassman** [comment]
Not really, sorry. I'm also not sure what you would consider to be useful tools. If this isn't related to Atom Commander in particular then try asking about this on the Atom's [discussion forum](https://discuss.atom.io/)

**Comment by sbaker7** [comment]
Jumping in just to say it would be really awesome to have chmod permissions as NJseo suggested. It should at least remember what the chmod permissions of the file was before save and apply the same settings. It's a bit frustrating having to go back to another program (ftp in terminal or filezilla, etc) after save to fix all the issues that have arisen because the chmod permissions have changed.

**Comment by morassman** [comment]
Hi @sbaker7 

I wasn't even aware that the permissions change after a save. Does this only happen when using FTP or does it happen when editing local files as well?

**Comment by sbaker7** [comment]
As far as I can tell, only ftp butI haven't tested it very thoroughly. It defaults to 640.

**Comment by morassman** [comment]
Thanks. I'll look into it for the next update.

**Comment by morassman** [comment]
I started working on this, but I haven't been able to reproduce it. I work on OSX and tested with FTP and SFTP, both of which are hosted locally.

I tested as follows:

1.  Create a test file : `test.sh`
2.  Change permissions to 666
3.  Edit the file via FTP and save.
4.  Checked the file permissions : Was still 666.
5.  Edit the file via SFTP and save.
6.  Checked the file permissions : Was still 666.
7.  Repeat the process with 777. Permissions remained 777 after each save.

I just want to make 100% sure. Are you using FTP or SFTP?

If it is SFTP then I can re-apply the permissions after each save. In my case this hasn't been necessary, but it may depend on the server being connected to whether it needs to be done or not.

**Comment by sbaker7** [comment]
Hi @morassman I'm sorry I haven't checked this in a while, I've been working around it by changing them in filezilla.

It is only ftp, definitely not sftp.
